nwbwidgets/image.py

```diff
@@ -1,94 +1,173 @@
-from pathlib import Path, PureWindowsPath
+from pathlib import Path
+from typing import Union
 
 import matplotlib.pyplot as plt
+import numpy as np
 import plotly.graph_objects as go
 import pynwb
-from ipywidgets import widgets, fixed, Layout
+from ipywidgets import widgets, Layout
 from pynwb.image import GrayscaleImage, ImageSeries, RGBImage
-from tifffile import imread, TiffFile
 
 from .base import fig2widget
-from .controllers import StartAndDurationController
+from .utils.cmaps import linear_transfer_function
+from .utils.imageseries import get_frame_count, get_frame
 from .utils.timeseries import (
     get_timeseries_maxt,
     get_timeseries_mint,
     timeseries_time_to_ind,
 )
 
+PathType = Union[str, Path]
+
 
 class ImageSeriesWidget(widgets.VBox):
     """Widget showing ImageSeries."""
 
     def __init__(
         self,
         imageseries: ImageSeries,
-        foreign_time_window_controller: StartAndDurationController = None,
-        **kwargs
+        foreign_time_slider: widgets.FloatSlider = None,
+        neurodata_vis_spec: dict = None,
     ):
         super().__init__()
         self.imageseries = imageseries
-        self.controls = {}
-        self.out_fig = None
+        self.figure = None
+        self.time_slider = foreign_time_slider
 
-        # Set controller
-        if foreign_time_window_controller is None:
-            tmin = get_timeseries_mint(imageseries)
-            if imageseries.external_file and imageseries.rate:
-                tif = TiffFile(imageseries.external_file[0])
-                tmax = imageseries.starting_time + len(tif.pages) / imageseries.rate
-            else:
-                tmax = get_timeseries_maxt(imageseries)
-            self.time_window_controller = StartAndDurationController(tmax, tmin)
+        if imageseries.external_file is not None:
+
+            # set time slider:
+            tmax = (
+                imageseries.starting_time
+                + get_frame_count(imageseries.external_file[0]) / imageseries.rate
+            )
+            if self.time_slider is None:
+                self.time_slider = widgets.FloatSlider(
+                    min=imageseries.starting_time,
+                    max=tmax,
+                    orientation="horizontal",
+                    description="time(s)",
+                )
+            external_file = imageseries.external_file[0]
```

Comment:
I think it would be easier if you set this as a class attribute

Reply:
Yes with callbacks now set as methods, it makes sense to have it as an attribute: https://github.com/NeurodataWithoutBorders/nwb-jupyter-widgets/commit/8e72ae96033c16009e718c501f7c62666f6ed2c8